packages/utils/src/utils/index.js

```diff
@@ -405,9 +410,16 @@ export function getTaskRunsWithPlaceholders({
   taskRuns,
   tasks
 }) {
-  const pipelineTasks = []
-    .concat(pipeline?.spec?.tasks)
-    .concat(pipelineRun?.spec?.pipelineSpec?.tasks)
+  let pipelineTasks = [];
+
+  if (pipelineRun?.status?.pipelineSpec?.tasks) {
+    pipelineTasks = pipelineTasks.concat(pipelineRun.status.pipelineSpec.tasks);
+  } else {
+    pipelineTasks = pipelineTasks
+      .concat(pipeline?.spec?.tasks)
+      .concat(pipelineRun?.spec?.pipelineSpec?.tasks);
+  }
+  pipelineTasks = pipelineTasks
     .concat(pipeline?.spec?.finally)
     .concat(pipelineRun?.spec?.pipelineSpec?.finally)
     .filter(Boolean);
```

Comment:
Do we need to add something similar for `finally` tasks referenced from a bundle?